src/components/ArtifactTimeline.vue

```diff
@@ -0,0 +1,98 @@
+<template>
+  <div class="artifact-timeline">
+    <p-virtual-scroller
+      :items="artifacts"
+      :item-estimate-height="112"
+      class="artifact-timeline__virtual-scroller"
+      @bottom="fetchMore"
+    >
+      <template #default="{ item: artifact }">
+        <ArtifactTimelineItem
+          v-bind="{ artifact }"
+          v-model:expanded="expanded"
```

Comment:
Curious why the list needs to know which items are expanded. It doesn't appear to be doing anything based on this. Could this just be internal to the items themselves?